packages/mdc-textfield/_mixins.scss

```diff
@@ -24,22 +24,37 @@
 @import "@material/line-ripple/mixins";
 @import "@material/notched-outline/mixins";
 @import "@material/theme/mixins";
+@import "@material/shape/mixins";
+@import "@material/shape/functions";
 @import "helper-text/mixins";
 @import "icon/mixins";
 @import "icon/variables";
 @import "./variables";
+@import "./functions";
 
-@mixin mdc-text-field-corner-radius($radius) {
-  border-radius: $radius $radius 0 0;
+@mixin mdc-text-field-shape-radius($radius, $rtl-reflexive: false) {
+  @if length($radius) > 2 {
+    @error "Invalid radius: '#{$radius}' component doesn't allow customizing all corners";
+  }
+
+  $masked-radius: mdc-shape-mask-radius($radius, 1 1 0 0);
+
+  @include mdc-shape-radius(mdc-shape-resolve-percentage-radius($mdc-text-field-height, $masked-radius), $rtl-reflexive);
 }
 
-@mixin mdc-text-field-textarea-corner-radius($radius) {
-  border-radius: $radius;
+@mixin mdc-text-field-textarea-shape-radius($radius, $rtl-reflexive: false) {
+  @if length($radius) > 2 {
```

Comment:
We shouldn't be restricting this mixin to 2 corners, at least as textarea is currently styled.

The masking was purposely performed only on the floating-label within the textarea before. This got changed in this PR specifically when you removed the masked-top-radius mixin.